src/DataProtection/Cryptography.KeyDerivation/src/PBKDF2/NetCorePbkdf2Provider.cs

```diff
@@ -54,11 +54,9 @@ private static byte[] DeriveKeyImpl(string password, byte[] salt, KeyDerivationP
                     throw new ArgumentOutOfRangeException();
             }
 
-            var passwordBytes = Encoding.UTF8.GetBytes(password);
-            using (var rfc = new Rfc2898DeriveBytes(passwordBytes, salt, iterationCount, algorithmName))
-            {
-                return rfc.GetBytes(numBytesRequested);
-            }
+            Span<byte> bytes = stackalloc byte[password.Length];
```

Comment:
Passwords could be arbitrarily large no? This might need a fall back to ArrayPool to avoid a StackOverflow. You'll also need to use `Encoding.Utf.GetMaxBytes(password.Length))` to get the maximum possible size since passwords could have unicode chars that need more than one byte to represent

Reply:
Updated to use stackalloc up to 256 bytes, and then shared array pool otherwise